libraries/standard/mqtt/integration-test/mqtt_system_test.c

```diff
@@ -593,18 +611,17 @@ static MQTTStatus_t unsubscribeFromTopic( MQTTContext_t * pContext,
                              globalUnsubscribePacketIdentifier );
 }
 
-
 static MQTTStatus_t publishToTopic( MQTTContext_t * pContext,
                                     const char * pTopic,
+                                    bool setRetainFlag,
```

Comment:
You know, if we're going to pass in all these parameters to this function, why not just pass in an `MQTTPublishInfo_t`? The only parts that aren't affected by the parameters here are `pPayload` and `payloadLength`, which this function could still set if you wanted.

Reply:
If we pass the `MQTTPublishInfo_t`, then this function wouldn't be required anymore!
The main utility that this function provides is commonizing the logic of assigning values to the `publishInfo` struct

Reply:
In that case, I'm wondering how much benefit this function actually provides any more, since every invocation of it uses almost as many lines as just setting the parameters of the publish info. Since a lot of these parameters are similar across different calls to this function, why don't we create a const `globalPublishInfoInitializer` to copy into a new struct each time, and only change the parameters we want to change for each publish?

Not a requirement for this PR, but a suggestion for how we can improve this for the future.

Reply:
Yes, I did think about that, but didn't want to increase the scope of this PR further 🙂